torch/_dynamo/output_graph.py

```diff
@@ -458,20 +461,59 @@ def call_user_compiler(self, gm):
                 else ""
             )
             _step_logger()(logging.INFO, f"calling compiler function {name}")
-            compiled_fn = self.compiler_fn(gm, self.example_inputs())
+
+            backend_correct = True
```

Comment:
I'm going to factor this (moving wrapper compiler inline) out into a standalone PR

Reply:
I also notice that you did some unrelated refactoring while you're in the code; I'm going to slavishly follow the original implementation, and you should feel free to reapply your refactor if you still want it.

Reply:
https://github.com/pytorch/pytorch/pull/89662

It doesn't pull the trigger on the API change but should make it simple to do so when we actually need it.